src/Microsoft.AspNetCore.Antiforgery/Internal/IAntiforgeryTokenGenerator.cs

```diff
@@ -20,9 +21,13 @@ public interface IAntiforgeryTokenGenerator
         /// Generates a request token corresponding to <paramref name="cookieToken"/>.
         /// </summary>
         /// <param name="httpContext">The <see cref="HttpContext"/> associated with the current request.</param>
+        /// <param name="principal">The claims-based principal to use for token generation.</param>
         /// <param name="cookieToken">A valid cookie token.</param>
         /// <returns>An <see cref="AntiforgeryToken"/>.</returns>
-        AntiforgeryToken GenerateRequestToken(HttpContext httpContext, AntiforgeryToken cookieToken);
+        AntiforgeryToken GenerateRequestToken(
+            HttpContext httpContext,
+            ClaimsPrincipal principal,
```

Comment:
Lots of hmming and hawing over here. Do see the slight cleanliness of not seeing `context.User` used. But I'm not sure it makes sense for this method to support as bizarre a scenario as generating a token for anyone but the current user. Can imagine future developers looking at this and trying to figure out a realistic scenario where passing both `httpContext` and `principal` makes sense.